conans/client/cmd/uploader.py

```diff
@@ -201,12 +201,11 @@ def upload(self, reference_or_pattern, remotes, upload_recorder, package_id=None
                                            all_packages, query)
 
         if parallel_upload:
-            self._upload_thread_pool = ThreadPool(8)
             self._user_io.disable_input()
-        else:
-            self._upload_thread_pool = ThreadPool(1)
 
         for remote, refs in refs_by_remote.items():
+            self._upload_thread_pool = ThreadPool(8 if parallel_upload else 1)
```

Comment:
Maybe this should be tools.cpu_count() instead?

Reply:
This doesn't look right either, as the ``self._upload_thread_pool`` variable will be overwritten, losing access to the other pool from the previous remote.

Reply:
Each iteration of this loop will complete before the next starts, since map() will block, then it will close the thread pool and start the next iteration.

Reply:
Wait, we can definitely reuse the thread pool here. Keeping it alive the whole time makes a lot more sense.